src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java

```diff
@@ -172,9 +174,17 @@ static String getNextFieldPlaceholder(Map<String, String> expressionAttributeNam
         if (start == -1) {
             return Optional.empty();
         }
-        int end = conditionExpression.indexOf(" ", start + toFind.length()); // -1
-        return Optional.of(conditionExpression.substring(start + toFind.length(),
-            end == -1 ? conditionExpression.length() : end)); // ":value" // TODO add support for non-EQ operators
+
+        String remainder = conditionExpression.substring(start + toFind.length());
+
+        int end = indexOfUsingRegex(" |,", remainder);
+
+        return Optional.of(end == -1 ? remainder : remainder.substring(0, end));
+    }
+
+    private static int indexOfUsingRegex(String regex, String str) {
```

Comment:
perhaps inline and make Pattern.compile(" |") a constant (rather than rebuilding it every time?)

Reply:
fixed in 7e766ac32745b44342100ca2002b70280815f90f